extensions/vault/runtime/src/main/java/io/quarkus/vault/VaultTransitSecretEngine.java

```diff
@@ -0,0 +1,95 @@
+package io.quarkus.vault;
+
+import java.util.List;
+
+import io.quarkus.vault.transit.ClearData;
+import io.quarkus.vault.transit.DecryptionRequest;
+import io.quarkus.vault.transit.DecryptionResult;
+import io.quarkus.vault.transit.EncryptedData;
+import io.quarkus.vault.transit.EncryptionRequest;
+import io.quarkus.vault.transit.EncryptionResult;
+import io.quarkus.vault.transit.RewrappingRequest;
+import io.quarkus.vault.transit.Signature;
+import io.quarkus.vault.transit.SigningBatchRequest;
+import io.quarkus.vault.transit.SigningResult;
+import io.quarkus.vault.transit.TransitContext;
+import io.quarkus.vault.transit.VerificationRequest;
+import io.quarkus.vault.transit.VerificationResult;
+import io.quarkus.vault.transit.Verifyable;
+
+public interface VaultTransitSecretEngine {
+
+    // encrypt
+
+    /**
+     * Encrypt a regular string with a vault key configured in the transit secret engine.
+     * equivalent to:
+     * {@code encrypt(keyName, clearData.getBytes("UTF-8"), Optional.empty());}
+     * <p>
+     * this method is usually used in conjunction with {@link #decrypt(String, String)}
+     * 
+     * @see <a href="https://www.vaultproject.io/api/secret/transit/index.html#encrypt-data">encrypt data documentation</a>
+     * @param keyName name of the transit vault key
+     * @param clearData the string to encrypt
+     * @return cipher text
+     */
+    String encrypt(String keyName, String clearData);
+
+    /**
+     * Encrypt a regular string with a vault key configured in the transit secret engine.
+     * if the key does not exist, and the policy specifies a create capability the key will be lazily created
+     * (i.e. upsert). the key can be further customized by specifying transit encryption-key configuration
+     * properties.
+     * 
+     * @see <a href="https://www.vaultproject.io/api/secret/transit/index.html#encrypt-data">encrypt data documentation</a>
+     * @param keyName name of the transit vault key
+     * @param clearData the data to encrypt
+     * @param transitContext an optional context for key derivation
+     * @return cipher text
+     */
+    EncryptedData encrypt(String keyName, ClearData clearData, TransitContext transitContext);
+
+    /**
+     * Encrypt a list of elements. This will return a list of results, each being either a ciphertext or an error.
+     * Each element can specify an optional context, allowing key derivation.
+     * 
+     * @see <a href="https://www.vaultproject.io/api/secret/transit/index.html#encrypt-data">encrypt data documentation</a>
+     * @param keyName name of the transit vault key
+     * @param requests the list of elements to encrypt
+     * @return a list of result objects containing either a ciphertext or an error
+     */
+    List<EncryptionResult> encrypt(String keyName, List<EncryptionRequest> requests);
+
+    // decrypt
+
+    String decrypt(String keyName, String encryptedData);
```

Comment:
This one should return `ClearData` as indeed you mentioned we don't know for sure which version of `encrypt` was used to get the data encrypted

Reply:
done